sim/main_game.py

```diff
@@ -3,40 +3,76 @@
 from settings.settings import Settings
 from characters.player.player import Player
 
+
 class projectG:
-    '''
+    """
     Class to manage game assets and behavior
-    '''
+    """
 
     def __init__(self):
-        '''
+        """
         Initialize game and create game resources
-        '''
+        """
         pygame.init()
 
         self.settings = Settings()
 
-        self.screen = pygame.display.set_mode((self.settings.screen_width,
-                                                self.settings.screen_height))
+        self.screen = pygame.display.set_mode(
+            (self.settings.screen_width, self.settings.screen_height)
+        )
         pygame.display.set_caption("ProjectG")
         self.player = Player(self)
 
-
     def run_game(self):
-        '''
+        """
         Start the main loop of the game
-        '''
+        """
         while True:
-            #Watch for keyboard and mouse events.
-            for event in pygame.event.get():
-                if event.type == pygame.QUIT:
-                    sys.exit()
+            # Watch for keyboard and mouse events.
+            self._check_events()
+            # Check Player movements
+            self.player.update()
+
+            # Make screen visible
+            self._update_screen()
+
+    def _check_events(self):
+        """
+        Responds to keypresses and mouse events
+        """
+
+        for event in pygame.event.get():
```

Comment:
Can be shorter with dict + setattr:

```
key_map = {
    pygame.K_RIGHT: 'moving_right',
    pygame.K_LEFT: 'moving_left',
    pygame.K_UP: 'moving_up',
    pygame.K_DOWN: 'moving_down',
}

for event in pygame.event.get():
    if event.type == pygame.QUIT:
        sys.exit()
    elif event.type in (pygame.KEYDOWN, pygame.KEYUP):
        movement = event.type == pygame.KEYDOWN
        if event.key in key_map:
            setattr(self.player, key_map[event.key], movement)
```

(have not verified this yet)

Reply:
This works!!!